include/wx/arrstr.h

```diff
@@ -47,6 +47,12 @@ wxDictionaryStringSortDescending(const wxString& s1, const wxString& s2)
     return wxDictionaryStringSortAscending(s2, s1);
 }
 
+int wxCMPFUNC_CONV wxNaturalStringSortAscending(const wxString& s1, const wxString& s2);
```

Comment:
I don't think it makes sense to call these functions "sort ascending/descending". The existing functions are solely called like this because they're used as predicates for the corresponding sort algorithms. Here we should have just 2 functions at most, with the second one being the `NoCase` version and not 4.

But actually I think we ought to have just a single function, `wxCmpStringsNaturally()` or something like this both because case-sensitive comparison is not very useful for the strings in the UI and because `StrCmpLogicalW()`, which I still think must be used under MSW, doesn't support case-sensitive comparison.

Reply:
The naming of the existing functions is indeed not perfect, but they're only meant to be used as `Sort()` callbacks. The new function might be useful for other things too, it's really more like `wxString::Cmp()` or `wxStrcmp()` and should have `Cmp` in its name somewhere.

Maybe `wxNaturalSortAscending()` and `wxNaturalSortDescending()` could be provided as trivial inline wrappers around it in the same way as it's done with `wxStringSortXXX()`, but the comparison function should still exist.